kerastuner/engine/hyperparameters.py

```diff
@@ -558,44 +558,61 @@ def register(self, name, type, config):
         if full_name in self.values:
+            return self.values[full_name]
+
+        # Check if current scope contains the conditional param. If found but
+        # not active, raise an error.
+        full_cond_name = self._get_name(name)
+        if full_cond_name in self.values:
             if self._conditions_are_active():
-                return self.values[full_name]
+                return self.values[full_cond_name]
+            elif default is not None:
+                return default
             else:
-                # Sanity check for conditional HP usage.
-                return None
+                raise ValueError(
+                    'Conditional parameter {} is not currently active'.format(
+                        full_cond_name))
 
-        # Check parent/child conditions.
+        # Check for any active conditional param.
         found_inactive = False
-        # Remove conditional scopes from name.
-        full_name_no_cond = '/'.join([
-            p for p in self._get_name_parts(full_name) if
-            not self._is_conditional_scope(p)])
-        for hp_name in self.values.keys():
-            hp_parts = self._get_name_parts(hp_name)
-            # Remove conditional scopes from name.
-            hp_name_no_cond = '/'.join(
-                [p for p in hp_parts
-                 if not self._is_conditional_scope(p)])
-            if hp_name_no_cond == full_name_no_cond:
-                # Check that this HP is active for this Trial.
-                if self._conditions_are_active(hp_parts):
-                    return self.values[hp_name]
+        for name, val in self.values.items():
+            hp_parts = self._get_name_parts(name)
+            hp_scopes = hp_parts[:-1]
+            hp_name = hp_parts[-1]
+            hp_full_name = self._get_name(
+                hp_name,
+                scopes=hp_scopes,
+                include_cond=False)
+            if full_name == hp_full_name:
+                if self._conditions_are_active(hp_scopes):
+                    return val
                 else:
                     found_inactive = True
 
-        if found_inactive:
-            # Sanity check, found only inactive conditional HPs.
-            return None
+        if default is not None:
+            return default
+        elif found_inactive:
+            raise ValueError(
+                'Conditional parameter {} is not currently active'.format(
+                    full_cond_name))
+        else:
+            raise ValueError(
+                'Unknown parameter: {}'.format(full_name))
 
-        raise ValueError(
-            'Unknown parameter: {}'.format(full_name_no_cond))
+    def __getitem__(self, name):
+        return self.get(name)
+
+    def __contains__(self, name):
+        # Valid hp values are only int, float, str.
+        indicator = self
+        item = self.get(name, default=indicator)
+        return item != indicator
```

Comment:
Seems like a surprising implementation; why not do a try/except around `get()`?

Reply:
Done